### PR TITLE
Switch from DelayedJob to Sidekiq

### PR DESCRIPTION
**Pull request by jcoyne**
Needs to have a redis hooked up in puppet.

## Why was this change made?

DelayedJob is not stopping jobs consistently (see #2104) and it is not consistent with the rest of our architecture.

## How was this change tested?

To be tested on stage

## Which documentation and/or configurations were updated?
Needs to have REDIS_URL set by puppet or shared configs.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -198,29 +198,7 @@ and in development or testing mode:
...
-00:00 delayed_job:status
-      01 ./bin/eye info delayed_job
-      01 command :info, objects not found!
-      01 command :info, objects not found!
-    ✘ 01 lyberadmin@argo-qa-a.stanford.edu 0.801s
-    ✘ 01 lyberadmin@argo-qa-b.stanford.edu 0.813s
-```
-
-This means the eye daemon was shutdown and did not restart. To start it back up, invoke `cap {ENV} delayed_job:reload`. Note that this is for reloading eye, not for restarting the workers. You should not have to invoke this task with any regularity; it is for recovering from unusual circumstances only.
+Argo uses systemd to manage and monitor its Sidekiq-based background job workers in all deployed environments.
```
Comment (mjgiarlo) [inline README.md:201]: How would you feel about including a link to https://github.com/sul-dlss/dlss-capistrano#sidekiq-via-systemd here?

File: app/jobs/modsulator_job.rb [hunk 2]
```diff
@@ -32,8 +31,7 @@ def perform(apo_id, uploaded_filename, output_directory, user, groups, filetype
     File.open(log_filename, 'a') do |log|
       start_log(log, user, original_filename, note)
 
-      # If a modsulator request fails, the job will fail and automatically be
-      # retried (see config/initializers/delayed_job.rb)
+      # If a modsulator request fails, the job will fail and automatically be  retried
```
Comment (mjgiarlo) [inline app/jobs/modsulator_job.rb:34]: Trivial niggle: there's an extra space between `be` and `retried`

File: lib/capistrano/tasks/delayed_job.rake [hunk 3]
```diff
@@ -1,77 +0,0 @@
-# frozen_string_literal: true
```
Comment (mjgiarlo) [inline lib/capistrano/tasks/delayed_job.rake:1]: :sob:

**Comment by mjgiarlo** [comment]
@jcoyne see https://github.com/sul-dlss/puppet/pull/5550. Feel free to un-draft and coordinate w/ @sul-dlss/operations on getting the merged so it's in sequence w/ merging/deploying this PR. Cool?

**Comment by jcoyne** [comment]
After I deployed this to stage, I had to manually kill eye and delayed_job.

**Comment by mjgiarlo** [comment]
> After I deployed this to stage, I had to manually kill eye and delayed_job.

Maybe run `cap ENV delayed_job:stop` immediately before deploying? No biggie either way. Thanks for letting us know.

**Pull request merged**